### ISSUE TITLE
CollectionEnvelope.java:43-47: CollectionEnvelope should...

### ISSUE THREAD

**Issue by 0pdd** [yegor256/cactoos#1184]
The puzzle `947-361be9e4` from #947 has to be resolved: 

https://github.com/yegor256/cactoos/blob/b092754a6e18a39951e27733490be1961dadfeb0/src/main/java/org/cactoos/collection/CollectionEnvelope.java#L43-L47

The puzzle was created by Victor Noël on 30-Jun-19. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) please, pay attention to this issue

**Comment by 0crat** [comment]
Job [#1184](https://github.com/yegor256/cactoos/issues/1184) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
The job [#1184](https://github.com/yegor256/cactoos/issues/1184) assigned to @marceloamadeu[/z](https://www.0crat.com/u/marceloamadeu), here is [why](https://www.0crat.com/footprint/C63314D6Z/e80f337f-f6a8-4404-ab7f-7bb6cfb0c6c8); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#1184` already assigned to @marceloamadeu, can't assign to @marceloamadeu

**Comment by victornoel** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/1184#issuecomment-541337610))

@victornoel This is what I know about this job in [C63314D6Z](https://www.0crat.com/p/C63314D6Z), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1184](https://github.com/yegor256/cactoos/issues/1184) is [in scope](http://www.zerocracy.com/policy.html#14) for 27days
  * The role is `DEV`
  * The job is assigned to @marceloamadeu[/z](https://www.0crat.com/u/marceloamadeu) for [24days](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 30 minutes/points
  * These users are banned and won't be assigned:
    * @0pdd[/z](https://www.0crat.com/u/0pdd): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%231184%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by victornoel** [comment]
@paulodamaso it seems @marceloamadeu is not reacting, any way to let it back in the open job pool so that it can be fixed? it's blocking another issue

**Comment by victornoel** [comment]
@paulodamaso ping

**Comment by paulodamaso** [comment]
@victornoel wanna fix it?

**Comment by victornoel** [comment]
@paulodamaso yep, can do :)

**Comment by paulodamaso** [comment]
@0crat assign @victornoel

**Comment by 0crat** [comment]
> @0crat assign @victornoel  ([here](https://github.com/yegor256/cactoos/issues/1184#issuecomment-544578109))

@paulodamaso The job [#1184](https://github.com/yegor256/cactoos/issues/1184) assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/e5c87e5b-7d7e-4cf9-a8ba-e30a881c8fb6); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); we should be aware that @victornoel[/z](https://www.0crat.com/u/victornoel) is on vacation; this ticket may be delayed; there will be no monetary reward for this job

**Comment by victornoel** [comment]
@0crat wait for #1210 to be merged first as it contains code changes to `CollectionEnvelope`

**Comment by 0crat** [comment]
> @0crat assign @victornoel  ([here](https://github.com/yegor256/cactoos/issues/1184#issuecomment-544578109))

@paulodamaso There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@bfb660cf-26fa-4ff3-bf0a-6258ad1da898, thread: PQ-C63314D6Z
com.zerocracy.claims.ClaimIn[214] java.lang.IllegalArgumentException: Parameter "age" not found in "Order was canceled" among: [comment, job, issue, repo, cause, _parent_sig, flow, login]
```

[0.50.30](https://github.com/zerocracy/farm): CID: [cb6e9096-431e-4890-9437-7ef44060997b](https://www.0crat.com/footprint/C63314D6Z/cb6e9096-431e-4890-9437-7ef44060997b), Type: "Order was canceled"

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
> @0crat wait for #1210 to be merged first as it contains code changes to `CollectionEnvelope` ([here](https://github.com/yegor256/cactoos/issues/1184#issuecomment-544610053))

@victornoel The impediment for [#1184](https://github.com/yegor256/cactoos/issues/1184) was registered successfully by @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by victornoel** [comment]
@0crat wait for #1216 to be merged

**Comment by 0crat** [comment]
> @0crat wait for #1216 to be merged ([here](https://github.com/yegor256/cactoos/issues/1184#issuecomment-549069028))

@victornoel Job [#1184](https://github.com/yegor256/cactoos/issues/1184) is already on hold

**Comment by 0pdd** [comment]
The puzzle `947-361be9e4` has disappeared from the source code, that's why I closed this issue.

**Comment by 0pdd** [comment]
@0pdd the puzzle [#1242](https://github.com/yegor256/cactoos/issues/1242) is still not solved.

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1184](https://github.com/yegor256/cactoos/issues/1184) is now out of scope

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0pdd** [comment]
@0pdd 14 puzzles [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305) are still not solved; solved: [#1242](https://github.com/yegor256/cactoos/issues/1242).

**Comment by 0pdd** [comment]
@0pdd 13 puzzles [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305) are still not solved; solved: [#1242](https://github.com/yegor256/cactoos/issues/1242), [#1294](https://github.com/yegor256/cactoos/issues/1294).

**Comment by 0pdd** [comment]
@0pdd 10 puzzles [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305) are still not solved; solved: [#1242](https://github.com/yegor256/cactoos/issues/1242), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1303](https://github.com/yegor256/cactoos/issues/1303).

**Comment by 0pdd** [comment]
@0pdd 10 puzzles [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320) are still not solved; solved: [#1242](https://github.com/yegor256/cactoos/issues/1242), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1303](https://github.com/yegor256/cactoos/issues/1303).

**Comment by 0pdd** [comment]
@0pdd 10 puzzles [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1325](https://github.com/yegor256/cactoos/issues/1325) are still not solved; solved: [#1242](https://github.com/yegor256/cactoos/issues/1242), [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1303](https://github.com/yegor256/cactoos/issues/1303).

**Comment by 0pdd** [comment]
@0pdd 9 puzzles [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326) are still not solved; solved: [#1242](https://github.com/yegor256/cactoos/issues/1242), [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304).

**Comment by 0pdd** [comment]
@0pdd 8 puzzles [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1325](https://github.com/yegor256/cactoos/issues/1325) are still not solved; solved: [#1242](https://github.com/yegor256/cactoos/issues/1242), [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1326](https://github.com/yegor256/cactoos/issues/1326).

**Comment by 0pdd** [comment]
@0pdd 7 puzzles [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320) are still not solved; solved: [#1242](https://github.com/yegor256/cactoos/issues/1242), [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326).

**Comment by 0pdd** [comment]
@0pdd 4 puzzles [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1320](https://github.com/yegor256/cactoos/issues/1320) are still not solved; solved: [#1242](https://github.com/yegor256/cactoos/issues/1242), [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326).

**Comment by 0pdd** [comment]
@0pdd 4 puzzles [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1337](https://github.com/yegor256/cactoos/issues/1337) are still not solved; solved: [#1242](https://github.com/yegor256/cactoos/issues/1242), [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326).

**Comment by 0pdd** [comment]
@0pdd 2 puzzles [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1337](https://github.com/yegor256/cactoos/issues/1337) are still not solved; solved: [#1242](https://github.com/yegor256/cactoos/issues/1242), [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326).

**Comment by 0pdd** [comment]
@0pdd 2 puzzles [#1337](https://github.com/yegor256/cactoos/issues/1337), [#1349](https://github.com/yegor256/cactoos/issues/1349) are still not solved; solved: [#1242](https://github.com/yegor256/cactoos/issues/1242), [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326).

**Comment by 0pdd** [comment]
@0pdd all 20 puzzles are solved here: [#1242](https://github.com/yegor256/cactoos/issues/1242), [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326), [#1337](https://github.com/yegor256/cactoos/issues/1337), [#1349](https://github.com/yegor256/cactoos/issues/1349).
